### PR TITLE
Use proper error pages instead of always redirecting

### PR DESCRIPTION
**Pull request by juliusknorr**
Implements a generic 404 error page instead of always redirecting to the login page or dashboard as discussed in https://github.com/nextcloud/server/issues/33048

Can be tested by browsing to an invalid url like `/index.php/apps/nonexisting`

Screenshot 2022-10-18 at 20 30 14

Added the `pending documentation` label as we need to adjust the nginx config in the docs accordingly to what was done in the htaccess.

### CODE REVIEW

**Comment by szaimen** [comment]
Does this close https://github.com/nextcloud/android/issues/10628#issuecomment-1217873647?

File: lib/base.php [hunk 1]
```diff
@@ -1069,15 +1070,28 @@ public static function handleRequest() {
...
+		}
+
+		try {
+			return OC::$server->get(\OC\Route\Router::class)->match('/error/404');
+		} catch (\Exception $e) {
+			logger('core')->emergency($e->getMessage(), ['exception' => $e]);
+			$l = \OC::$server->getL10N('lib');
+			OC_Template::printErrorPage(
+				$l->t('404'),
+				$l->t('We can\'t find the page you\'re looking for.'),
```
Comment (szaimen) [inline lib/base.php:1092]:
```suggestion
				$l->t('We cannot find the page you are looking for.'),
```

**Comment by juliushaertl** [comment]
> Does this close https://github.com/nextcloud/android/issues/10628#issuecomment-1217873647?

It would help but doesn't explain why the redirect is happening. The 404 should already be returned by the dav app, so it might just hide another issue. But I cannot reproduce this case either. Maybe it is related to the general webserver configuration (besides .htaccess)?

**Review by CarlSchwan — APPROVED** [review]
:heart_eyes: 

Finally the current behavior was driving me insane some times!

**Review by skjnldsv — APPROVED** [review]

**Comment by szaimen** [comment]
autoloaders not up-to-date

**Comment by juliushaertl** [comment]
Nginx config does't seem to have this anymore since a long time:

https://github.com/nextcloud/documentation/pull/392/files

Does anyone have a nginx setup at hand to check if this is properly working there?

**Comment by DaphneMuller** [comment]
hello @juliushaertl , 
Thank you for your work on this pull request! This ticket seems to have the tag 'missing documentation', is there any chance you could clarify what documentation is missing? Is this for admins or for app developers?

**Comment by juliushaertl** [comment]
Documentation is already up to date, but I've added a note for administrators with steps to take in https://github.com/nextcloud/server/issues/34692

**Pull request merged**